test/AWS.Deploy.CLI.IntegrationTests/SaveCdkDeploymentProject/SaveCdkDeploymentProjectTests.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.IO;
+using System.Linq;
+using AWS.Deploy.CLI.Extensions;
+using AWS.Deploy.CLI.IntegrationTests.Extensions;
+using AWS.Deploy.CLI.IntegrationTests.Services;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace AWS.Deploy.CLI.IntegrationTests.SaveCdkDeploymentProject
+{
+    public class SaveCdkDeploymentProjectTests : IDisposable
```

Comment:
We are doing FileIO in tests which is discouraged. Can you use `IDirectoryManager` and `IFileManager` mocks? Our test cases must not rely on the FileIO. We already have test implementations for both. You can extend then based on your needs.

Reply:
The `IDirectoryManager` and `IFileManager` mocks are used to **bypass** any File I/O that is encountered while testing a larger scheme of things. In this case File I/O itself is the core component that we need to test. Trying to write unit tests which fakes the behavior of File I/O does not achieve our purpose and that is why I have included integration tests to perform end-to-end testing. I also spoke about this with @normj and he concurs.